### ISSUE TITLE
MNIST example does not work

### ISSUE THREAD

**Issue by RuABraun** [xu-ji/IIC#12]
I've been trying to run the unsupervised MNIST example. I took the command from [here](https://github.com/xu-ji/IIC/blob/master/examples/commands.txt), and adjusted `dataset_root` and `out_root` appropriately.

But when I run it the program hangs, and to stop it I have to restart the computer (no response from ctrl c/z). Htop shows it's using one core for only kernel calls, the longest I've let it run is one hour. I tried looking into the code but haven't figured out what is going wrong. Is there an additional parameter one should change before calling the command?

Side question: What are normal loss values when using the IIC loss function?

**Comment by xu-ji** [comment]
Hi, I just reran the script for MNIST on my machine and it runs fine. 

1. Do `git pull` to make sure you have the latest code
2. Try to run a smaller job. E.g:

`export CUDA_VISIBLE_DEVICES=3 && nohup python -m code.scripts.cluster.cluster_greyscale_twohead --model_ind 0 --arch ClusterNet6cTwoHead --mode IID --dataset MNIST --dataset_root /scratch/local/ssd/xuji/MNIST --gt_k 10 --output_k_A 10 --output_k_B 10  --lamb_A 1.0 --lamb_B 1.0 --lr 0.0001 --num_epochs 3200 --batch_sz 10 --num_dataloaders 1 --num_sub_heads 1 --crop_orig --crop_other --tf1_crop centre_half --tf2_crop random --tf1_crop_sz 20  --tf2_crop_szs 16 20 24 --input_sz 24 --rot_val 25 --no_flip --head_B_epochs 2 > out/sh10_gpu3_m0.out &`

This will not give meaningful results but just to check the code runs for you. (This one only takes 580MB GPU memory for me.)

If you download the trained models you can see all the numbers and plots. E.g. for MNIST model 685 for head B (main head) the loss goes from -1.22 to -2.20.

**Comment by RuABraun** [comment]
I managed to reproduce results on MNIST by writing it from scratch (with the loss function copied over), and don't have time to track down the cause of the problems I had, so feel free to close this if you want to.
